### ISSUE TITLE
3DS mode support (YANBF merge/integration)?

### ISSUE THREAD

**Issue by Yrouel** [MechanicalDragon0687/ndsForwarder#31]
Given that YANBF exists now, can the creation of such 3DS forwarders instead of DSiWare be added?

This would make ndsForwarder the ultimate forwarder: no more finicky java crap (accomplished by ndsForwarder), no more having to move the microsd to the computer (annoying), no more 40 titles limit (accomplished by YANBF) and easy custom banner creation (accomplished by YANBF).

In other words it would seem the perfect marriage of these two projects

**Comment by MechanicalDragon0687** [comment]
Firstly, absolutely, yes. I have stepped out of the 3DS development and Nintendo Homebrew community for the foreseeable future but this is still something i would like to do.

A couple considerations since you may not be aware of how these things function.  

YANBF uses existing PC software to perform the functions it needs to perform.  Unfortunately, that software is not available on the 3ds.  
The current nds forwarder generator app was effectively a rewrite of existing software for the 3ds. In order to do a 3ds version of the forwarders, the app must rewrite the PC software again.  3DS apps are more complex than DSiWare software. This is not something that is easy to do and takes a significant time investment.

I do plan to do this eventually if nobody else does, but as it stands it will likely be a year or longer before I attempt to start rewriting it.

**Comment by Yrouel** [comment]
Ah thanks for the extra info, I was hoping it was somewhat easier to implement but I can see it's not

**Comment by lifehackerhansol** [comment]
Yeah, we had this discussion, for now we've left it alone. I myself am fully aware of how horribly incompatible the structure is with on-device generation.

The code is actually relatively simple to straight-port to C, however the problem is that the 3DS doesn't have any proper form of image manipulation library which I rather abuse in the Python code. So even after a rewrite in C actually using it on-console will still be very annoying to manipulate. 
On DSiWare forwarders, there was a really simple answer: have a template CIA and literally just copy paste the banner binary (icon, title and all) where the forwarder CIA expects it. But I didn't have this luxury on YANBF...

So yeah, it'll be a while before we see something.
